telethon/tl/custom/dialog.py

```diff
@@ -96,6 +96,12 @@ def __init__(self, client, dialog, entities, messages):
         return await self._client.send_message(
             self.input_entity, *args, **kwargs)
 
+    async def delete(self):
+        if self.is_channel:
+            await self._client(functions.channels.LeaveChannelRequest(self.input_entity))
+        else:
```

Comment:
I guess we need an `elif self.is_group: functions.messages.DeleteChatUserRequest(self.entity.id, types.InputPeerSelf())` here.

Reply:
To remove the exited group from dialogs : `functions.messages.DeleteHistoryRequest(self.entity.id, 0, just_clear=False)`